### PR TITLE
Refactor validation rule, organising files into packages and handling some error cases

### PR DESCRIPTION
**Pull request by Vishnuviki**

### CODE REVIEW

**Comment by Vishnuviki** [comment]
Hi there, you can review the changes based on each commit, so that will be easy. I organised the files into different packages. You can also change it think if needed

**Comment by Vishnuviki** [comment]
This is enough, I will create another PR for the test and other changes, this is done.

**Comment by Vishnuviki** [comment]
Hi @yatzek, I know this is a fat PR, but this solves the organising of the files and packages and there are no functional changes. If you have a chance to look at it, please review and approve it. I will create another PR for test cases.

File: controllers/main_controller.go [hunk 1]
```diff
@@ -43,29 +45,35 @@ func (mc *MainController) ValidateRule(c *gin.Context) {
 		session.Save()
 		c.Redirect(http.StatusFound, "/")
 		return
-	}
-
-	if !data.IsDestinationAddressValid() {
+	} else if !data.IsDestinationAddressValid() {
```
Comment (yatzek) [inline controllers/main_controller.go:48]: unnecessary "else" should be omitted, see effective go: https://go.dev/doc/effective_go#if

File: controllers/main_controller.go [hunk 2]
```diff
@@ -43,29 +45,35 @@ func (mc *MainController) ValidateRule(c *gin.Context) {
...
+		// TODO: what happen if user types `slack.com`??
+		// We need to change the logic of checking FQDN
 		session.AddFlash("Validation failed: Destination address not valid, must be IP or FQDN")
 		session.Save()
 		c.Redirect(http.StatusFound, "/")
 		return
+	} else {
```
Comment (yatzek) [inline controllers/main_controller.go:55]: unnecessary "else" should be omitted, see effective go: https://go.dev/doc/effective_go#if

Comment (Vishnuviki) [reply controllers/main_controller.go:55]: I changed this logic in my another PR, will create once this merged

**Pull request merged**